src/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -31,6 +31,9 @@
   <PropertyGroup Condition="'$(TargetsUnix)'=='true'">
     <DefineConstants>PLATFORM_UNIX;$(DefineConstants)</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetsOSX)'=='true'">
+    <DefineConstants>PLATFORM_OSX;$(DefineConstants)</DefineConstants>
```

Comment:
If we are going to have different builds for OSX (I think it is a fine idea), we should switch the existing runtime detection of OSX to be static instead: 
https://github.com/dotnet/corert/blob/fe8995999d52af7c73b3e44eb3a4e7fbbe8bd7c3/src/System.Private.CoreLib/src/System/IO/Path.Unix.cs#L17